pgaudit.c

```diff
@@ -984,6 +988,19 @@ log_select_dml(Oid auditOid, List *rangeTabls)
         Oid relOid;
         Relation rel;
         RangeTblEntry *rte = lfirst(lr);
+        
+        /* 
+         * We only show the table in the query, 
+         * there is no need to show partitions or inheritance child. 
+         */        
+        if ( inheritedExists == true && rte->requiredPerms == ACL_NO_RIGHTS)
+            continue;
+         
+        /* used for inherited table and partitioned table*/
+        if( rte->inh == true)
+        {
+            inheritedExists = true;           
+        }        
```

Comment:
Removing lines 995-1003 does not change the test output. What are you trying to accomplish here?

Reply:
the purpose of these code is to remove the partition and inheritance child log when query related to partioned table or inherited table.
set the pgaudit.log = 'ALL' and we could find the difference.

Run a test scenario as below to compare the result.
step-1: show the partition table and inherited table
postgres=# \dt
                List of relations
 Schema |  Name   |       Type        |  Owner   
--------+---------+-------------------+----------
 public | h       | partitioned table | postgres
 public | h_0     | table             | postgres
 public | h_1     | table             | postgres
 public | person  | table             | postgres
 public | student | table             | postgres


step-2: select from the partitioned table and inherite table
postgres=# select * from h;
Jun 16 23:09:24 localhost postgres[20171]: [30-1] LOG:  AUDIT: SESSION,12,1,READ,SELECT,PARTITIONED TABLE,public.h,select * from h;,<not logged>

postgres=# select * from person;
Jun 16 23:20:22 localhost postgres[20330]: [16-1] LOG:  AUDIT: SESSION,3,1,READ,SELECT,TABLE,public.person,select * from person;,<not logged>


step-3: I comment the code, the result is not correct. The log shows as below.
Actually, the user input is "select * from partitioned table", there is no need to show partitions.
similary, the user input is "select * from inherited table", there is no need to show inheritance child.

postgres=# select * from h;
Jun 16 23:11:23 localhost postgres[20244]: [16-1] LOG:  AUDIT: SESSION,3,1,READ,SELECT,PARTITIONED TABLE,public.h,select * from h;,<not logged>
Jun 16 23:11:23 localhost postgres[20244]: [17-1] LOG:  AUDIT: SESSION,3,1,MISC,UNKNOWN,TABLE,public.h_0,<previously logged>,<previously logged>
Jun 16 23:11:23 localhost postgres[20244]: [18-1] LOG:  AUDIT: SESSION,3,1,MISC,UNKNOWN,TABLE,public.h_1,<previously logged>,<previously logged>

postgres=# select * from person;
Jun 16 23:17:59 localhost postgres[20244]: [21-1] LOG:  AUDIT: SESSION,6,1,READ,SELECT,TABLE,public.person,select * from person;,<not logged>
Jun 16 23:17:59 localhost postgres[20244]: [22-1] LOG:  AUDIT: SESSION,6,1,MISC,UNKNOWN,TABLE,public.person,<previously logged>,<previously logged>
Jun 16 23:17:59 localhost postgres[20244]: [23-1] LOG:  AUDIT: SESSION,6,1,MISC,UNKNOWN,TABLE,public.student,<previously logged>,<previously logged>

Reply:
OK, but commenting out the code does not cause any test failures so at the very least there are missing tests. Also, I'm not convinced the code is quite right. Even if it is I'm not sure it can be back-patched.

What I propose is that we commit the changes I have made to this branch to fix only the incorrect class for partitioned tables/indexes and then consider the additional filtering in a separate PR, that may or may not be back-patched.

Let me know what you think.

Reply:
Hi David, I agree with you to just fix the incorrect object type issue in this branch.
I will have a further study on additional filtering with a seperate PR.
Thanks David, appreciate with your careful and responsible attitude.